pkg/serialization/codec/jam/decode.go

```diff
@@ -395,7 +443,27 @@ func (br *byteReader) decodeBytes(dstv reflect.Value) error {
 	dstv.Set(reflect.ValueOf(b).Convert(inType))
 	return nil
 }
+func (br *byteReader) decodeBits(v *[]bool) (err error) {
+	length := len(*v)
+	if length > math.MaxUint32 {
+		return ErrExceedingByteArrayLimit
+	}
+	var b byte
+	for i := range *v {
+		mod := i % 8
+		if mod == 0 {
+			b, err = br.ReadOctet() // take as many bytes as needed to fill the bit sequence
+			if err != nil {
+				return err
+			}
+		}
+		pow2 := byte(1 << mod)   // powers of 2
+		(*v)[i] = b&pow2 == pow2 // identify the bit
```

Comment:
Is this ensuring least significant bit extraction?

Reply:
the least significant bit will be `b&1<<0` when i % 8 == 0